protocol/x/vault/keeper/withdraw.go

```diff
@@ -42,3 +44,100 @@ func (k Keeper) ValidateWithdrawFromVault(
 
 	return nil
 }
+
+func (k Keeper) RedeemShares(
```

Comment:
big nit: in terms of naming, should we just use "withdraw" everywhere instead of "redeem" to be consistent